### DISCUSSION TITLE
Suggestions

### DISCUSSION THREAD

**Discussion by Popeye1001** [majianjia/QingStation#4] [Ideas]
1. What do you think about this sensor?
https://www.instructables.com/Sonic-Anemometer/
Is it easier to build and precise?

2. Maybe a vibrating rain sensor?
https://www.aqara.com/us/vibration_sensor.html

3. I recommend a humidity sensor with a heater
BME680 or SHT30-DIS and SPL06-001

**Comment by majianjia** [comment]
1. It is very interesting. I knew some anemometer has significant less gap between the reflection pad. I didn't know the principle until you show this project. I think it is even simpler to build than the one I presented. 
But the circuit he proposed is based on the hardware phase detection, which then relies on the accuracy of the timer in the microcontroller.
It might be difficult to achieve a higher frequency.  I will still recommend using ADC then use digital signal processing.
In the sampling, it requires at least 3 independent ADC peripherals (not channels) to sample at the same time. Need some high-end MCU. 
(I suppose 1ADC with multi-channels also works but you will need to measure/estimate the minor delays between channels.)

The problem with my current design is it requires at least 5 CM of gaps, but my signal is too weak. 
In high winds, it will fail to recognize the beam. 

For the driving, the array itself, the one you post is easier than the one I built. 
It does not require analogue switches. and it got only one active transducer in the middle, all other three are passive. 
All these lead to an even simpler analog circuit. 

Thanks for your suggestion, I think I will build a new one use this array layout. 
If you are also very interested in this, we can have further discussion. 

2. Vibration rain sensor is also a new thing to me. The current optical one is quite hard to calibrate, and it is affected by the sunlight. 
We have IMU on board but I don't think it is easy to differ between raindrops and wind.

3. I didn't use BME680 because I can't buy it at the time. It was a bad idea.

**Comment by Popeye1001** [comment]
https://m.media-amazon.com/images/I/71dbmgMtraL._AC_SL1500_.jpg

For ADC mayby ADS1220 (1 to 128 gain, 24 bit, i2c) ?

Transducers will be better with 30 degrees or 45 degrees direction?

**Reply by majianjia** [reply]
I think speed is more important than accuracy. But I didn't understand his circuit and principle, so there might be some misunderstanding.

**Comment by Popeye1001** [comment]
Now I'm waiting for the rain to test the vibration sensor.
I have a UV sensor under construction with JGS1.
I will share my results soon. I leave the anometer issues to you.
Take a look at the Ecowitt WS80 is powered by 2 AA batteries. Maybe a 3.3V power rail is enough
